sklearn/linear_model/_ridge.py

```diff
@@ -1720,7 +1720,7 @@ def _eigen_decompose_covariance(self, X, y, sqrt_sw):
             cov[:, -1] = 0
             cov[-1, -1] = sqrt_sw.dot(sqrt_sw)
         nullspace_dim = max(0, n_features - n_samples)
-        eigvals, V = linalg.eigh(cov)
+        eigvals, V = linalg.eigh(cov, check_finite=False)
```

Comment:
`cov[-1, -1]` is the dot product of two weight vectors, which can be non-finite, even when the input is finite:

```python
import numpy as np

f_max = np.finfo(np.float64).max
sqrt_sw = np.full((10), 0.9 * f_max)

print(np.isfinite(sqrt_sw).all())
# True

cov11 = sqrt_sw.dot(sqrt_sw)

print(np.isfinite(cov11).all())
# False
```